fe/fe-core/src/main/java/org/apache/doris/nereids/rules/RuleType.java

```diff
@@ -23,18 +23,22 @@
 public enum RuleType {
     // binding rules
     BINDING_UNBOUND_RELATION_RULE,
+    BINDING_SENTINEL,
```

Comment:
We should use a enum RuleClass to avoid sentinel. e.g.
```java
enum RuleClass {
  BIND_RULE, REWRITE_RULE,  EXPLORATION_RULE...
}

public enum RuleType {
   BINDING_UNBOUND_RELATION_RULE(BIND_RULE),
   LOGICAL_JOIN_COMMUTATIVE(EXPLORATION_RULE)
   ...
}
```